src/Message/Cache/MessageCacheCreator.php

```diff
@@ -94,44 +100,61 @@ private function rebuildCache(array $lookupPathList, ConfigCache $configCache):
         }
 
         $map = var_export($messageMap, true);
-        $configCache->write("<?php\nreturn {$map};");
+        
+        $hash = (new MessageCacheHashCalculator())->calculateHash($files);
+        
+        $configCache->write("<?php\nreturn {$map};", [new MessageCacheResource($hash)]);
     }
 
     /**
      * @param array<string> $lookupPathList
+     * @param array<string> $excludePathList
      * @return array<string>
      */
-    private function loadPhpFilesFromPathList(array $lookupPathList): array
+    private function loadPhpFilesFromPathList(array $lookupPathList, array $excludePathList): array
     {
         $files = [];
 
         foreach ($lookupPathList as $lookupPath) {
-            $files = array_merge($files, $this->recursivePhpFileSearch($lookupPath));
+            $files = array_merge($files, $this->recursivePhpFileSearch($lookupPath, $excludePathList));
         }
 
         return $files;
     }
 
     /**
      * @param string $path
+     * @param array<string> $excludePathList
      * @return array<string>
      */
-    private function recursivePhpFileSearch(string $path): array
+    private function recursivePhpFileSearch(string $path, array $excludePathList): array
     {
         $pattern = $path . '/*.php';
         $files = glob($pattern);
 
         if ($files === false) {
             return [];
         }
+        
+        if (count($excludePathList) > 0) {
```

Comment:
Ich fände es besser direkt am Anfang der Methode zu schauen ob $path in der $excludePathList vorkommt. Dadurch würde sowohl der lookup im FS entfallen als auch das array_filter für alle betroffenen Ordner

Reply:
Ist denke ich nicht sinnvoll. Das würde ja heißen jemand gibt einen Pfad an der durchsucht werden soll und excluded diesen dann direkt wieder. In der Regel wird es so sein, dass man als Pfad ja dann den kompletten src Ordner angibt und darin dann nochmal einzelne Ordner mit einzelnen Dateien excludet. Das lässt sich ja im Vorfeld  so nicht ausschließen. Davon abgesehen dürfte die Zeitersparnis auch nicht groß sein. Das was hier teuer ist, ist das parsen des Codes selbst und nicht das Listen der Dateien.

Reply:
Ich denke das das sehr wohl Sinnvoll ist. Erstens weil diese Methode rekursiv durchlaufen wird und damit nicht unbedingt der Ursprungspfad wieder ausgeschlossen wird. Zweitens: Deutlich besser lesbar. Drittens macht es rein logisch keinen sinn jede datei einzeln zu prüfen wenn eh der gesamte Pfad ausgeschlossen ist...